src/registry/utils.ts

```diff
@@ -0,0 +1,97 @@
+import {
```

Comment:
Another utils file. Uhg. But this is the type of thing I want to "hide" from the main implementations, so here it is... in utils.ts...